### PR TITLE
MRG: converting from Ximg to Nifti

### PR DESCRIPTION
**Pull request by matthew-brett**
Have image manipulation down, but need some basic work to get header data from
attrs, and less basic work to calculate affine in the general case.

### CODE REVIEW

**Comment by matthew-brett** [comment]
@effigies - would you mind taking a look at this?  It implements the version of your suggestion over at https://github.com/pydata/xarray/discussions/8897#discussioncomment-8988009 . See the `with_updated_affines` method in the `xi` accessor.

**Comment by matthew-brett** [comment]
@ivanov @effigies - do either of you have any idea what is going on here: https://github.com/matthew-brett/xibabel/actions/runs/8558868993/job/23454329975#step:10:13  ?  The tests run fine on macOS and Linux, and they run fine on a local Windows machine as well.   There aren't any stray unicode characters that I can detect, I've retyped the line with no recorded changes in `git diff`, now I'm a bit stuck.

File: src/xibabel/loaders.py [hunk 1]
```diff
@@ -620,3 +633,119 @@ def guess_format(self, file_path):
...
+            out_affines[space] = self._adjusted_affine(affine, adj_img.coords)
+        adj_img.attrs['xib-affines'] = out_affines
+        return adj_img.xi._with_reset_coordinates()
+
+    def _adjusted_affine(self, affine, coords):
+        vox_origin = np.zeros(3)
+        vox_scalings = np.ones(3)
+        for i, name in enumerate(_NI_SPACE_DIMS):
+            vox_indices = np.atleast_1d(coords.get(name, 0))
+            vox_origin[i] = vox_indices[0]
```
Comment (effigies) [inline src/xibabel/loaders.py:731]: This only works if it's also at most 1D. Is that guaranteed?

**Review by effigies — COMMENTED** [review]
Read the loaders.py code and it looks like what I had in mind. Haven't yet had a chance to read the tests to verify the usage looks as expected.

Doing a lot of traveling this week and next, so my notifications are a mess. Please bug me if I don't revisit this.

**Pull request merged**